File: magellan-library/src/main/java/com/wealthfront/magellan/ScreenLifecycleListener.java
package com.wealthfront.magellan;

/**
 * Use {@link Navigator#addLifecycleListener(ScreenLifecycleListener)} to add a lifecycle listener that will be called
 * whenever a Screen's onShow or onHide method is called.
 */
public interface ScreenLifecycleListener {

  void onShow(Screen screen);

  void onHide(Screen screen);

}

File: magellan-kotlin/src/main/java/com/wealthfront/magellan/kotlin/ScreenLifecycleListener.kt
package com.wealthfront.magellan.kotlin

/**
 * Use [Navigator.addLifecycleListener] to add a lifecycle listener that will be called
 * whenever a Screen's onShow or onHide method is called.
 */
interface ScreenLifecycleListener {

  fun onShow(screen: Screen<*>)

  fun onHide(screen: Screen<*>)

}